src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +472,54 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private object GenerateWatchData(MirrorData data)
+        {
+            if (data.IsCollection)
+            {
+                var list = new List<object>();
+                var elements = data.GetElements();
+
+                foreach (var mirrorData in elements)
+                {
+                    if (mirrorData.IsCollection)
+                    {
+                        list.Add(GenerateWatchData(mirrorData));
+                    }
+                    else if (mirrorData.Data != null)
+                    {
+                        double number;
+                        list.Add(double.TryParse(mirrorData.StringData, out number)
+                            ? mirrorData.Data
+                            : mirrorData.Data.ToString());
+                    }
+                    else
+                    {
+                        list.Add(mirrorData.Class == null
+                            ? "null"
+                            : mirrorData.Class.ClassName);
```

Comment:
Any reason not to use, ToString?  Most data types in Dynamo should provide sensible results here.